feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,737 @@
+
+			cases := []struct {
+				desc          string
+				counter       ygnmi.SingletonQuery[uint64]
+				parentCounter ygnmi.SingletonQuery[uint64]
+			}{
+				{
+					desc:          "InDiscards",
+					counter:       counters.InDiscards().State(),
+					parentCounter: parentCounters.InDiscards().State(),
+				},
+				{
+					desc:          "InErrors",
+					counter:       counters.InErrors().State(),
+					parentCounter: parentCounters.InErrors().State(),
+				},
+				{
+					desc:          "InUnknownProtos",
+					counter:       counters.InUnknownProtos().State(),
+					parentCounter: parentCounters.InUnknownProtos().State(),
+				},
+				{
+					desc:          "OutDiscards",
+					counter:       counters.OutDiscards().State(),
+					parentCounter: parentCounters.OutDiscards().State(),
+				},
+				{
+					desc:          "OutErrors",
+					counter:       counters.OutErrors().State(),
+					parentCounter: parentCounters.OutErrors().State(),
+				},
+				{
+					desc:          "InFcsErrors",
+					counter:       counters.InFcsErrors().State(),
+					parentCounter: parentCounters.InFcsErrors().State(),
+				},
+			}
+
+			for _, c := range cases {
+				t.Run(c.desc, func(t *testing.T) {
+					if val, present := gnmi.Lookup(t, dut, c.counter).Val(); present && (val == 0) {
+						t.Logf("INFO: %s: %d", c.counter, val)
+					} else if pVal, pPresent := gnmi.Lookup(t, dut, c.parentCounter).Val(); pPresent && (pVal == 0) {
+						t.Logf("INFO: %s: %d", c.parentCounter, val)
+					} else {
+						t.Logf("ERROR: %s and %s check fail", c.counter, c.parentCounter)
+					}
+				})
+			}
+		}
+	}
+}
+
+func TestInterfaceEthernetNoDrop(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		counters := gnmi.OC().Interface(intf).Ethernet().Counters()
+		cases := []struct {
+			desc    string
+			counter ygnmi.SingletonQuery[uint64]
+		}{
+			{
+				desc:    "InCrcErrors",
+				counter: counters.InCrcErrors().State(),
+			},
+			{
+				desc:    "InMacPauseFrames",
+				counter: counters.InMacPauseFrames().State(),
+			},
+			{
+				desc:    "OutMacPauseFrames",
+				counter: counters.OutMacPauseFrames().State(),
+			},
+			{
+				desc:    "InBlockErrors",
+				counter: counters.InBlockErrors().State(),
+			},
+		}
+
+		for _, c := range cases {
+			t.Run(c.desc, func(t *testing.T) {
+				if val, present := gnmi.Lookup(t, dut, c.counter).Val(); present && (val == 0) {
+					t.Logf("INFO: %s: want 0, got %d", c.counter, val)
+				} else {
+					t.Errorf("ERROR: %s: present %t, got %d, want 0", c.counter, present, val)
+				}
+			})
+		}
+	}
+}
+
+func TestSystemAlarms(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	query := gnmi.OC().System().AlarmAny().State()
+	alarms := gnmi.GetAll(t, dut, query)
+	if len(alarms) > 0 {
+		for _, a := range alarms {
+			alarmSeverity := a.GetSeverity()
+			// Checking for major system alarms
```

Comment:
period at end of comment

Reply:
Thanks for your comments. Added period for comment:
```go
			alarmSeverity := a.GetSeverity()
			// Checking for major system alarms.
			if alarmSeverity == oc.AlarmTypes_OPENCONFIG_ALARM_SEVERITY_MAJOR {
```